dev-client/src/screens/LocationDashboardScreen.tsx

```diff
@@ -43,21 +45,38 @@ export const LocationDashboardScreen = ({siteId, coords}: Props) => {
     coords = site!;
   }
 
-  const appBarRightButton = useMemo(
-    () =>
-      siteId ? (
-        <AppBarIconButton
-          name="settings"
-          onPress={() => navigation.navigate('SITE_SETTINGS', {siteId})}
-        />
-      ) : (
+  const userRole = useSelector(state =>
+    siteId === undefined ? null : selectUserRoleSite(state, siteId),
+  );
+
+  const appBarRightButton = useMemo(() => {
+    // display add button if no site associated with location
+    if (!siteId) {
+      return (
         <AppBarIconButton
           name="add"
           onPress={() => navigation.navigate('CREATE_SITE', {coords})}
         />
-      ),
-    [siteId, coords, navigation],
-  );
+      );
+    }
+
+    // display nothing if user does not owner the site / is not manager
+    if (
+      userRole === null ||
+      (userRole.kind === 'project' && userRole.role !== 'manager') ||
+      (userRole.kind === 'site' && userRole.role !== 'owner')
```

Comment:
comment: if not in this PR this logic should be encapsulated somewhere in a re-usable way